app_dart/lib/src/service/scheduler.dart

```diff
@@ -322,17 +325,54 @@ class Scheduler {
       commitSha: commitSha,
       reason: reason,
     );
-    final Commit presubmitCommit = Commit(branch: branch, repository: slug.fullName, sha: commitSha);
-    final List<LuciBuilder> presubmitBuilders = await getPresubmitBuilders(
-      commit: presubmitCommit,
-      prNumber: prNumber,
-    );
-    await luciBuildService.scheduleTryBuilds(
-      builders: presubmitBuilders,
-      slug: slug,
-      prNumber: prNumber,
-      commitSha: commitSha,
+    final github.CheckRun ciValidationCheckRun = await githubChecksService.githubChecksUtil.createCheckRun(
+      config,
+      slug,
+      'ci.yaml validation',
+      commitSha,
     );
+    dynamic exception;
+    final Commit presubmitCommit = Commit(branch: branch, repository: slug.fullName, sha: commitSha);
+    try {
+      final List<LuciBuilder> presubmitBuilders = await getPresubmitBuilders(
+        commit: presubmitCommit,
+        prNumber: prNumber,
+      );
+      await luciBuildService.scheduleTryBuilds(
+        builders: presubmitBuilders,
+        slug: slug,
+        prNumber: prNumber,
+        commitSha: commitSha,
+      );
+    } catch (e) {
+      exception = e;
+    }
+
+    // Update validate ci.yaml check
+    if (exception == null) {
+      // Success in validating ci.yaml
+      await githubChecksService.githubChecksUtil.updateCheckRun(
```

Comment:
We can add this to the end of the try block and remove the if - then else.

Reply:
This is separate to minimize the code the try block deals with (only scheduler config validation + schedule builds), otherwise we would silently fail on quota issues